examples/hello_world/module/rust/src/lib.rs

```diff
@@ -18,31 +18,65 @@ pub mod proto {
     include!(concat!(env!("OUT_DIR"), "/oak.examples.hello_world.rs"));
 }
 
+use either::Either;
 use log::info;
-use oak::{grpc, io::ReceiverExt};
+use oak::grpc;
 use oak_abi::{label::Label, proto::oak::application::ConfigMap};
+use oak_io::Sender;
 use proto::{HelloRequest, HelloResponse, HelloWorld, HelloWorldDispatcher};
 
-oak::entrypoint!(oak_main<ConfigMap> => |_receiver| {
-    oak::logger::init_default();
-    let node = Node {
-        translator: grpc::client::Client::new(
+#[derive(Default)]
+struct Main;
+
+oak::entrypoint_command_handler!(oak_main => Main);
+
+impl oak::CommandHandler for Main {
+    type Command = ConfigMap;
+
+    fn handle_command(&mut self, _command: Self::Command) -> anyhow::Result<()> {
+        let translator_sender_result = oak::io::node_create::<grpc::Invocation>(
             "translator",
+            &Label::public_untrusted(),
             &oak::node_config::wasm("translator", "oak_main"),
-            &Label::public_untrusted()
-        )
-        .map(translator_common::TranslatorClient),
-    };
-    let dispatcher = HelloWorldDispatcher::new(node);
-    let grpc_channel =
-        oak::grpc::server::init("[::]:8080").expect("could not create gRPC server pseudo-Node");
-    oak::run_command_loop(dispatcher, grpc_channel.iter());
-});
+        );
+        let handler_init_sender = oak::io::entrypoint_node_create::<HelloWorldDispatcher<Node>>(
+            "handler",
+            &Label::public_untrusted(),
+            "app",
+        )?;
+        let handler_command_sender = oak::io::send_init(
+            handler_init_sender,
+            translator_sender_result
+                .map(Either::Right)
+                .unwrap_or(Either::Left(())),
+            &Label::public_untrusted(),
+        )?;
+        oak::grpc::server::init_with_sender("[::]:8080", handler_command_sender)?;
+        Ok(())
+    }
+}
 
 struct Node {
     translator: Option<translator_common::TranslatorClient>,
 }
 
+impl oak::WithInit for Node {
+    // Since `Option<T>` is not directly `Encodable` and `Decodable`, we emulate it via a
+    // `Either<(), T>`, which is isomorphic to it.
+    type Init = Either<(), Sender<grpc::Invocation>>;
```

Comment:
Should we use `InitWrapper` instead of `Either`, so it will just be a struct with `.init` rather that `left()` or `right()` ?

Reply:
The point is the optionality, not the init aspect. How would `InitWrapper` help?

Reply:
I just thought that we already have a struct for sending init messages alongside commands:
https://github.com/project-oak/oak/blob/bd8ecb84ab8a4788243a73d5a2daf39782bd71f6/oak_io/src/lib.rs#L90-L94

We might use it everywhere

Reply:
I am not sure I follow.

`InitWrapper` contains one non-optional value, plus a receiver for other values.

What we need here is only a way of passing a handle, or nothing at all.

What is your suggestion for how to fill in the associated type to make it work the way it does now?

```rust
impl oak::WithInit for Node {
    type Init = InitWrapper<grpc::Invocation, grpc::Invocation>;
```

something like this?

Reply:
Also note that `InitWrapper` is a product type (i.e. a struct), but `Either` is a sum type (i.e. an enum), so I don't think there is a way of using them isomorphically (but let me know if I am mistaken).

Reply:
Ah, so this message gets wrapped in the `InitWrapper` already. I think I just got confused that multiple usages of the word `Init`

Reply:
But I still don't get why `Either` even though `left()` is never used?

Reply:
Correct, InitWrapper wraps Init + Receiver<Command>, then Init is handles by this trait, and Command is handled by CommandHandler.

Either<(), T> is isomorphic to Option<T> (if you think about it, Option is defined even in Rust as either None (i.e. `()`) or Some(T) (i.e. `T` itself), I am using it because Option<T> is not Encodable / Decodable. Makes sense?

Reply:
In this (isomorphic) case: why would you need `Option` if `None` is never used ?

Reply:
It is used below:

```rust
    fn create(init: Self::Init) -> Self {
        Node {
            translator: init // Either<(), T>
                .right() // Option<T>
                .map(|invocation_sender| grpc::client::Client { invocation_sender }) // Option<T>::map
                .map(translator_common::TranslatorClient),
        }
    }
```

I added some comments to clarify what the types are in the chain, does it help?

Reply:
The entire chain will evaluate to `None` if the value was `Either::Left(())`.